src/protocol/ipa_prf/prf_sharding/mod.rs

```diff
@@ -158,40 +143,40 @@ impl InputsRequiredFromPrevRow {
             record_id,
             &input_row.is_trigger_bit,
             &ever_encountered_a_source_event,
-            &tv,
+            &input_row.trigger_value,
             attribution_window_seconds,
-            &timestamp,
+            &input_row.timestamp,
             &source_event_timestamp,
         )
         .await?;
 
-        let updated_sum = self
-            .saturating_sum
-            .add(
-                ctx.narrow(&Step::ComputeSaturatingSum),
-                record_id,
-                &attributed_trigger_value,
-            )
-            .await?;
+        let (is_saturated, updated_sum) = integer_sat_add(
+            ctx.narrow(&Step::ComputeSaturatingSum),
+            record_id,
+            &self.saturating_sum,
+            &attributed_trigger_value,
+        )
+        .await?;
 
         let (is_saturated_and_prev_row_not_saturated, difference_to_cap) = try_join(
-            updated_sum.is_saturated.multiply(
-                &(share_of_one - &self.saturating_sum.is_saturated),
+            is_saturated.multiply(
+                &self.is_saturated.neg(),
                 ctx.narrow(&Step::IsSaturatedAndPrevRowNotSaturated),
                 record_id,
             ),
-            updated_sum.truncated_delta_to_saturation_point(
+            integer_sub(
                 ctx.narrow(&Step::ComputeDifferenceToCap),
                 record_id,
-                TV::BITS,
-            ),
+                &Replicated::<TV>::expand(Boolean::ZERO),
+                &updated_sum,
+            )
```

Comment:
I think that this will do the same thing as what I had before...

The only time that I need to use the "difference_to_cap" is if this row is the one that put us over the limit. So I know that the delta to cap is going to fit in the number of bits used to express a trigger value (no need for the larger quantity used for the cumulative sum).